pkg/app/api/grpcapi/piped_api.go

```diff
@@ -1269,3 +1259,17 @@ func (a *PipedAPI) validateEnvBelongsToProject(ctx context.Context, envID, proje
 	}
 	return nil
 }
+
+func (a *PipedAPI) updateDeploymentRefStatusIfNecessary(ctx context.Context, deploymentChainID string, blockIndex uint32, deploymentID string, status model.DeploymentStatus, reason string) error {
```

Comment:
super nit: how about putting this helper function at L542? I would prefer to place private functions close to the functions that call them.

But it's up to you.

Reply:
sure, addressed by https://github.com/pipe-cd/pipe/pull/2864/commits/82f8cc74849ced8bcaa1e1278869bb5a9da68e20 🙏